ssl/ssl_lib.c

```diff
@@ -4118,6 +4169,42 @@ SSL_CTX *SSL_CTX_new_ex(OSSL_LIB_CTX *libctx, const char *propq,
         goto err;
     }
 
+#ifndef OPENSSL_NO_SSLKEYLOG
+    /* Make sure we have a global lock allocated */
+    if (!RUN_ONCE(&ssl_keylog_once, ssl_keylog_init)) {
+        /* log an error, but don't fail here */
+        ERR_raise(ERR_LIB_SSL, ERR_R_SSL_LIB);
+    } else {
+        /* If someone set SSLKEYLOGFILE ... */
+        if (keylogfile != NULL) {
```

Comment:
Minor nit but if you invert these to check `keylogfile != NULL` first before `!RUN_ONCE` you can avoid needing to instantiate the global state for processes which don't have `SSLKEYLOGFILE` set. You might also want to handle `strlen(keylogfile) == 0` too.

Reply:
refactored